src/pybind/mgr/diskprediction/agent/predict/prediction.py

```diff
@@ -0,0 +1,308 @@
+                    predicted = int(disk_info['prediction']['predicted'])
+                if disk_info.get('prediction', {}).get('near_failure'):
+                    if disk_info['prediction']['near_failure'] == 'Good':
+                        life_expectancy_day_min = (TIME_WEEK * 6) + TIME_DAYS
+                        life_expectancy_day_max = None
+                    elif disk_info['prediction']['near_failure'] == 'Warning':
+                        life_expectancy_day_min = (TIME_WEEK * 2)
+                        life_expectancy_day_max = (TIME_WEEK * 6) - TIME_DAYS
+                    elif disk_info['prediction']['near_failure'] == 'Bad':
+                        life_expectancy_day_min = 0
+                        life_expectancy_day_max = (TIME_WEEK * 2) - TIME_DAYS
+                    else:
+                        # Near failure state is unknown.
+                        predicted = None
+                        life_expectancy_day_min = None
+                        life_expectancy_day_max = None
+                check_dev_id = ''
+                if model:
+                    check_dev_id += str(model).upper()
+                if serial_number:
+                    check_dev_id += (" %s" % serial_number).upper()
+                check_dev_id = check_dev_id.replace(' ', '_')
+                for dev_n, dev_info in devs_info.iteritems():
+                    if dev_info.get('dev_id', '').find(check_dev_id) >= 0:
+                        dev_id = dev_info['dev_id']
+                        break
+
+                if predicted and dev_id and life_expectancy_day_min:
+                    from_date = None
+                    to_date = None
+                    try:
+                        if life_expectancy_day_min:
+                            from_date = datetime.datetime.fromtimestamp(predicted/(1000**3)+life_expectancy_day_min).strftime('%Y-%m-%d')
+                        if life_expectancy_day_max:
+                            to_date = datetime.datetime.fromtimestamp(predicted/(1000**3)+life_expectancy_day_max).strftime('%Y-%m-%d')
+                        obj_api.set_device_life_expectancy(dev_id, from_date, to_date)
+                        self._logger.info("succeed to set device %s life expectancy from: %s, to: %s" % (dev_id, from_date, to_date))
+                    except Exception as e:
+                        self._logger.error("failed to set device %s life expectancy from: %s, to: %s, %s" % (dev_id, from_date, to_date, str(e)))
+                else:
+                    if dev_id:
+                        obj_api.reset_device_life_expectancy(dev_id)
+
+            effected = self._effected_resource(osd_id)
+            collect_data.update({
+                'host': hostname,
+                'last_time': int(time.time()),
+                osd_key: {
+                    'id': osd_id,
+                    'uuid': osd_uuid,
+                    'physicalDisks': info_list,

```

Comment:
If this key is under your control, then consider to use ``physical_disks`` instead as mentioned here: https://www.python.org/dev/peps/pep-0008/?#function-and-variable-names

Reply:
@votdev Thanks.